src/python/pants/util/memo.py

```diff
@@ -24,6 +26,44 @@ def equal_args(*args, **kwargs):
   return key
 
 
+def coercing_arg_normalizer(type_coercions, base_normalizer, args, kwargs):
```

Comment:
Seems this can be _private, fwict you only intend to export `coerce_collections` and `coerce_collections_per_instance`.